sys/compat/cheriabi/cheriabi_misc.c

```diff
@@ -1954,8 +1954,7 @@ cheriabi_ptrace(struct thread *td, struct cheriabi_ptrace_args *uap)
 		c.pl.pl_syscall_code = r.pl.pl_syscall_code;
 		c.pl.pl_syscall_narg = r.pl.pl_syscall_narg;
 		memcpy(c.pl.pl_tdname, r.pl.pl_tdname, sizeof(c.pl.pl_tdname));
-		siginfo_native_to_siginfo(&r.pl.pl_siginfo,
-		    (_siginfo_t *)&c.pl.pl_siginfo);
+		memcpy(&c.pl.pl_siginfo, &r.pl.pl_siginfo, sizeof(c.pl.pl_siginfo));
```

Comment:
We could add static_asserts that both variables are of the same type or have the same sizeof()?

Reply:
Actually, I hadn't realized it at the time, but c.pl and r.pl are identical and we could actually retire struct ptrace_lwpinfo_c entirely.  However, CHERIABI is going to be removed in the not-too-distant future and I have been avoiding making larger changes (like removing types) on the assumption we will just be removing sys/compat/cheriabi entirely once the native purecap ABI is stable.

Reply:
If they are the same type we could just do an assignment instead of memcpy?

Reply:
Well, one is siginfo_c and one is siginfo, but they are the same.  If it is too confusing I could start retiring the 'foo_c' types instead.

Reply:
Actually, we can't retire the foo_c types because they are embedded in the logic used by makesyscalls.lua (and having foo_c types there is an all-or-nothing thing, not something you can do piecemeal).  We could retire them perhaps once all the native types are gone, but not until then.